dash/include/dash/internal/Math.h

```diff
@@ -107,7 +107,7 @@ double sigma(const Iter& begin, const Iter& end)
  * \code
  *   long number  = 2 * 2 * 2 * 4 * 7 * 7;
  *   auto factors = dash::math::factorize(number);
- *   // returns map { (2,3), (4,1), (7,2) }
+ *   // returns map { (2,5), (7,2) }
  * \endcode
  */
 template <typename Integer>
```

Comment:
Add a `static_assert` that `Integer` is indeed an integral type?

Reply:
It should also be `unsigned`? And the exponents too?

Reply:
IMO it's better to add a runtime assert for positive numbers to avoid casting a negative number to `unsigned`. The factors should be unsigned though, yes.

Reply:
`static_assert` is compile time

Reply:
I think I got it:

* [ ] add `static_assert` to check that `Integer` is an integral type
* [ ] add `assert` that `n => 0`
* [ ] change mapped type from `int` to `unsigned`

correct?

Reply:
That's what I meant, yes :)